### PR TITLE
Unitary parametrization by rotations

### PR DESCRIPTION
**Pull request by araza6**
Fixes https://github.com/qgrad/qgrad/issues/54

### CODE REVIEW

**Comment by araza6** [comment]
This PR is review-ready now

File: qgrad/qgrad_qutip.py [hunk 1]
```diff
@@ -377,3 +380,102 @@ def to_dm(state):
         )
 
     return out
+
+
+def make_rot(N, params, idx):
```
Comment (quantshah) [inline qgrad/qgrad_qutip.py:385]: Maybe we can make this a private function, so this should be `_make_rot`. It is a bit confusion to the user if this is a public method.

File: qgrad/qgrad_qutip.py [hunk 2]
```diff
@@ -377,3 +380,102 @@ def to_dm(state):
...
+        
+    .. note::
+        There are a total of :math:`\frac{N}(N-1)}{2}` 
+        :math:`\theta_{ij}` parameters :math:`\frac{N}(N-1)}{2}` 
+        :math:`\phi{ij}` parameters, and :math:`N omega_{ij}`
+        parameters 
+        
+    Args:
+        N (int): Dimension of the unitary matrix
+        thetas (:obj:`jnp.ndarray`): theta angles for rotations
```
Comment (quantshah) [inline qgrad/qgrad_qutip.py:447]: This is well written. Maybe in order to be consistent with the other major operator in this library `Displace()`, we should go for a `Unitary()` class which has the same signature as displace. You construct a N dim unitary as `u = Unitary(N)` first, and then apply it such that `u(thetas, phis, omegas)` uses the `__call__` method to give the unitary.

**Comment by quantshah** [comment]
This is nicely done. The tests are also well written. To be consistent with the other operators (Displace), let us keep them class based (see my comment). It is a simple change, rest is fine for merging.

**Comment by araza6** [comment]
Thanks! I am wondering how would one provide a class-based API in this case? Because the unitary initialization needs both the rotation parameters for sin and cos and diagonal parameters for exponents at initialization. Also, when we would change the parameters, we would need to reconstruct rotation matrices, so that does not save us the `index_update` too

**Comment by araza6** [comment]
Perhaps I can merge this for now and try a unitary learning tutorial with this parameterization. We can discuss how a class-based `Unitary(N)` would work (as I highlight above), and I'll open a separate PR.
